client/router.js

```diff
@@ -0,0 +1,134 @@
+/*global window NODE_ENV*/
+import queryString from 'query-string';
+import extend from 'extend';
+
+import BaseRouter from 'shared/router'
+import { createAction } from 'redux-act';
+
+const updateLocation = createAction('Default action for updating url.');
+
+export { updateLocation };
+
+
+const DEFAULT_UPDATE_LOCATION_ACTION = {
+  type: updateLocation.getType(),
+  payload: {}
+};
+
+export default class Router extends BaseRouter {
+
+  get locale(){
+    return Router.locale();
+  }
+
+  initializeHistory(createHistory, store) {
+    let router = this;
+    router.history = createHistory();
+    router.history.listen(router.onLocationChange.bind(router, store));
+  }
```

Comment:
We should also accept a third parameter here so client can add an additional listener (for instance to install Google Analytics):

```js
  initializeHistory(createHistory, store, fnChange) {
     let router = this;
     router.history = createHistory();
     router.history.listen(router.onLocationChange.bind(router, store));
    if (fnChange) router.history.listen(fnChange);
   }
```